client/pkg/fileutil/purge.go

```diff
@@ -58,21 +57,14 @@ func purgeFile(lg *zap.Logger, dirname string, suffix string, max uint, interval
 			defer close(donec)
 		}
 		for {
-			fnames, err := ReadDir(dirname)
+			fnamesWithSuffix, err := readDirWithSuffix(dirname, suffix)
 			if err != nil {
 				errC <- err
 				return
 			}
-			newfnames := make([]string, 0)
-			for _, fname := range fnames {
-				if strings.HasSuffix(fname, suffix) {
-					newfnames = append(newfnames, fname)
-				}
-			}
-			sort.Strings(newfnames)
```

Comment:
Where is sorting?

Reply:
Yes, I did remove the sort operation after suffix filtering. I should have mentioned that in the PR description.

As I understand, the provided `ReadDir` reads directory filenames in sorted order. Sorted order should be preserved under suffix filtering, I believe? Unless there's an edge case I'm missing.

Reply:
> the provided `ReadDir` reads directory filenames in sorted order.

Indeed, 

https://github.com/etcd-io/etcd/blob/5a53a708d8ab9ef936ac5b8062ffc66c77a2c18f/client/pkg/fileutil/read_dir.go#L58

Reply:
Updated the PR description to mention that explicit sorting after suffix filtering is removed.